File: 2022_c#_basics/13-nested-loops/04-sum-of-two-numbers.cs
using System;

namespace _04_Sum_of_Two_Numbers
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int n1 = int.Parse(Console.ReadLine());
            int n2 = int.Parse(Console.ReadLine());
            int n3 = int.Parse(Console.ReadLine());

            int check = 0;
            bool isN3 = false;

            for (int i = n1; i <= n2; i++)
            {
                for (int j = n1; j <= n2; j++)
                {
                    check++;

                    if (i + j == n3)
                    {
                        isN3 = true;
                        Console.WriteLine($"Combination N:{check} ({i} + {j} = {n3})");
                        break;
                    }
                }

                if(isN3)
                {
                    break;
                }
            }

            if (!isN3)
            {
                Console.WriteLine($"{check} combinations - neither equals {n3}");
            }
        }
    }
}

File: 2024_java_basics/13-nested-loops/tasks/04-sum-of-two-numbers.java
import java.util.Scanner;

public class SumOfTwoNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n1 = scanner.nextInt();
        int n2 = scanner.nextInt();
        int n3 = scanner.nextInt();
        
        int check = 0;
        boolean isN3 = false;

        for (int i = n1; i <= n2; i++) {
            for (int j = n1; j <= n2; j++) {
                check++;
                
                if (i + j == n3) {
                    isN3 = true;
                    System.out.printf("Combination N:%d (%d + %d = %d)%n", check, i, j, n3);
                    break;
                }
            }
            
            if (isN3) {
                break;
            }
        }
        
        if (!isN3) {
            System.out.printf("%d combinations - neither equals %d%n", check, n3);
        }

        scanner.close();
    }
}
